core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -1026,42 +1026,16 @@ public void close() {
     statusTransitioner.close().succeeded();
   }
 
-  @Override
-  public Maintainable toMaintenance() {
-    final StatusTransitioner.Transition st = statusTransitioner.maintenance();
-    try {
-      final Maintainable maintainable = new Maintainable() {
-        @Override
-        public void create() {
-          Ehcache.this.create();
-        }
-
-        @Override
-        public void destroy() {
-          Ehcache.this.destroy();
-        }
-
-        @Override
-        public void close() {
-          statusTransitioner.exitMaintenance().succeeded();
-        }
-      };
-      st.succeeded();
-      return maintainable;
-    } catch (RuntimeException e) {
-      st.failed(e); // this throws
-    }
-    throw new AssertionError("Should not reach this line... ever!");
-  }
-  
-  void create() {
+  void checkMaintenance() {
     statusTransitioner.checkMaintenance();
-    // TODO figure out persistence and user managed caches
   }
 
-  void destroy() {
-    statusTransitioner.checkMaintenance();
-    // TODO figure out persistence and user managed caches
+  StatusTransitioner.Transition internalToMaintenance() {
```

Comment:
I think we talked about this offline, but I'll reiterate here.  This feels really funky, it seems wrong that the Ehcache class is exposing methods that allow for the direct manipulation of an internal state.  Also it seems wrong that the Ehcache class can be transitioned to a "Maintainable" state when it isn't even capable (on it's own) of being persistent.  It feels like we're bleeding details of the lifecycle of a persistent cache in to the non-persistent version.  Maybe the answer is an abstract superclass which contains methods to query the lifecycle but that the persistent and non persistent subclasses implement their own independent lifecycles?  That idea probably sucks... maybe this is the best solution.  The status quo seems weird to me though.